src/main/java/org/terasology/metalrenegades/minimap/CharacterOverlay.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.metalrenegades.minimap;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.geom.Rect2f;
+import org.terasology.math.geom.Rect2fTransformer;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector2f;
+import org.terasology.metalrenegades.ai.component.CitizenComponent;
+import org.terasology.minimap.overlays.MinimapOverlay;
+import org.terasology.rendering.assets.texture.Texture;
+import org.terasology.rendering.nui.Canvas;
+import org.terasology.utilities.Assets;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * This class is used to add character overlays to the minimap based on the citizen type
+ * and position
+ */
+
+
+public class CharacterOverlay implements MinimapOverlay {
+    private static final float ICON_SIZE = 16f;
+
+    private Vector2f iconSize = new Vector2f(ICON_SIZE, ICON_SIZE);
+
+    private Logger logger = LoggerFactory.getLogger(CharacterOverlay.class);
+
+    private ArrayList<EntityRef> Citizens;
+
+
+    /**
+     * This constructor sets creates the list of citizens
+     *
+     */
+
+
+    public CharacterOverlay(){
+        this.Citizens = new ArrayList<EntityRef>();
+    }
+
+    @Override
+    public void render(Canvas canvas, Rect2f worldRect) {
+
+        Collection<EntityRef> citizens = this.Citizens;
+
+
+        for (EntityRef CitizenEntity : citizens) {
+            if (!CitizenEntity.hasComponent(CitizenComponent.class)) {
+                logger.error("No Citizen found!");
+                return;
+            }
+
+            Rect2f screenRect = Rect2f.createFromMinAndSize(
+                    new Vector2f(canvas.getRegion().minX(), canvas.getRegion().minY()),
+                    new Vector2f(canvas.getRegion().maxX(), canvas.getRegion().maxY())
+            );
+
+            Rect2fTransformer transformer = new Rect2fTransformer(worldRect, screenRect);
```

Comment:
Don't create the `screenRect` and `transformer` inside the citizen entity loop because they don't depend on the citizen entity in question